### DISCUSSION TITLE
About MicroProfile and Quarkus

### DISCUSSION THREAD

**Discussion by albertowar** [quarkusio/quarkus#27974] [Q&A] [answered]
Hi Quarkus friends!

I'm currently exploring MicroProfile and I was happy to see that there is a Quarkus implementation in the [quarkiverse](https://github.com/quarkiverse/quarkus-microprofile).

I was wondering if this is something that Quarkus plans to support and extend in the future (MP 5.0).

I'm also curious to hear any success stories with MP + Quarkus that you may have.

Thanks!

**Comment by gsmet** [comment]
In fact, Eclipse MicroProfile is supported at the core of Quarkus: you have a lot of SmallRye extensions which are implementations of Eclipse MicroProfile.

AFAICS https://github.com/quarkiverse/quarkus-microprofile is not up to date so I would recommend that you use the individual SmallRye extensions in the core of Quarkus. See https://code.quarkus.io/?extension-search=origin:platform%20smallrye .

@radcortez do you actively maintain the Quarkiverse extension? I'm not sure I see its value? If you want to maintain it, it probably needs an update, if you don't then we should probably archive it.

**Reply by radcortez** [reply]
I maintain it, but I usually only update the version when we break the Quarkus compatibility. Maybe it was not a good idea to tie the version to the Quarkus version.

Also, we need it to comply with the rules to run the TCKs and claim compatibility.

**Comment by radcortez** [comment, marked as answer]
Yes, MP is core to Quarkus, and the core team implements the support.

MP 5 is only the Jakarta package rename (no feature differences, only `javax` to `jakarta`). As you can imagine, making Quarkus full Jakarta is a huge task (which is ongoing right now). When that is ready, then we will also move to MP 5.0.

**Reply by albertowar** [reply]
Thanks for your response!

How can we keep up to date with the progress of the Jakarta package rename?

**Reply by radcortez** [reply]
Please follow the progress here:
https://github.com/orgs/quarkusio/projects/13/views/20

We also have a Jakarta local build to try:
https://github.com/quarkusio/quarkus/tree/main/jakarta#build-locally
